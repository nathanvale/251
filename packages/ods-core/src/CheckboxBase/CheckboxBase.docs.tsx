/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { getSwitchBaseDocs } from "../_private/components/AbstractSwitch/switch-docs";
import { CheckboxBase, CheckboxBaseProps } from "./CheckboxBase";

export const docs: ComponentDocs<CheckboxBaseProps> = {
  category: "Internal",
  componentName: "CheckboxBase",
  description: [
    "This is a basic checkbox component without any labels or helper text.",
    "Use this atomic component to build more complicated components.",
    "The Checkbox components uses this under the hood.",
  ].join(" "),
  propDescriptions: getSwitchBaseDocs("checkbox"),
  migrationGuide: false,
  examples: [
    {
      label: "Uncontrolled",
      description: "Simple use case of an uncontrolled checkbox",
      Code: () => <CheckboxBase id="tc-checkbox" />,
    },
    {
      label: "Controlled",
      description: "It becomes controlled when checked prop is provided",
      Code: () => <CheckboxBase id="tc-checkbox" checked />,
    },
    {
      label: "Error",
      description: "Turns red when it is in error state and it is not checked",
      Code: () => <CheckboxBase id="tc-checkbox" error />,
    },
    {
      label: "Disabled",
      description: "All turn grey when it is disabled",
      Code: () => <CheckboxBase id="tc-checkbox" disabled checked />,
    },
  ],
  snippets: [
    {
      label: "Uncontrolled",
      Code: () => <CheckboxBase id="checkbox1" />,
    },
  ],
};
