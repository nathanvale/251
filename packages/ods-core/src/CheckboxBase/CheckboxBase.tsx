import React from "react";
import MuiCheckbox from "@material-ui/core/Checkbox";
import CheckBoxOutlineBlankSharpIcon from "@material-ui/icons/CheckBoxOutlineBlankSharp";
import CheckBoxSharpIcon from "@material-ui/icons/CheckBoxSharp";
import { CheckboxProps as MuiCheckboxProps } from "@material-ui/core";
import { SvgIconProps } from "@material-ui/core/SvgIcon/SvgIcon";
import { BaseInputProps } from "@origin-digital/ods-types";
import { makeStyles } from "@material-ui/core/styles";

export interface CheckboxBaseProps extends BaseInputProps {
  checked?: boolean;
  error?: boolean;
  "aria-describedby"?: string;
  muiProps?: MuiCheckboxProps;
}

const iconBaseStyle = {
  border: "none",
  boxShadow: "none",
};

const useUnheckedStyle = makeStyles({
  root: {
    ...iconBaseStyle,
  },
});

const UncheckedIcon = ({
  disabled,
  error,
  ...others
}: SvgIconProps & { disabled?: boolean; error?: boolean }) => {
  const classes = useUnheckedStyle();

  return (
    <CheckBoxOutlineBlankSharpIcon
      {...others}
      classes={classes}
      color={disabled ? "disabled" : error ? "error" : "inherit"}
    />
  );
};

const useCheckedStyle = makeStyles({
  root: {
    ...iconBaseStyle,
  },
});

const CheckedIcon = ({
  disabled,
  ...others
}: SvgIconProps & { disabled?: boolean }) => {
  const classes = useCheckedStyle();

  return (
    <CheckBoxSharpIcon
      {...others}
      classes={classes}
      color={disabled ? "disabled" : "inherit"}
    />
  );
};

const useCheckboxStyle = makeStyles({
  root: {
    border: "none",
    boxShadow: "none",
    alignItems: "flex-start",
    "&&:hover": {
      boxShadow: "none",
    },
  },
});

export const CheckboxBase = (props: CheckboxBaseProps) => {
  const {
    disabled,
    error,
    "aria-describedby": describedBy,
    muiProps,
    ...other
  } = props;
  const chkClasses = useCheckboxStyle();

  return (
    <MuiCheckbox
      {...muiProps}
      {...other}
      inputProps={{
        ...(muiProps && muiProps.inputProps),
        "aria-describedby": describedBy,
      }}
      disabled={disabled}
      checkedIcon={<CheckedIcon disabled={disabled} />}
      icon={<UncheckedIcon disabled={disabled} error={error} />}
      classes={chkClasses}
    />
  );
};

CheckboxBase.displayName = "CheckboxBase";
CheckboxBase.defaultProps = {
  "data-id": "checkboxBase",
};
