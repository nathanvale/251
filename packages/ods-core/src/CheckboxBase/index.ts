export { CheckboxBase } from "./CheckboxBase";
export type CheckboxBaseProps = import("./CheckboxBase").CheckboxBaseProps;
