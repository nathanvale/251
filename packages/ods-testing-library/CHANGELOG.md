# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
