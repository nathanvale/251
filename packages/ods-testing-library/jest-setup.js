import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
