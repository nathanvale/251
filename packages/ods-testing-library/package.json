{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
