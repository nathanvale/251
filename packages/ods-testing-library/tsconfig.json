{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist/typings",
    "rootDir": "./src"
  },
  "include": [
    "src",
    "src/**/*.json",
    "../../globals.d.ts",
    "../../@types",
    "jest-setup.ts"
  ],
  "references": [{"path": "../ods-themes/tsconfig.json"}]
}
