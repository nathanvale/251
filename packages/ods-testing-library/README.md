# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
