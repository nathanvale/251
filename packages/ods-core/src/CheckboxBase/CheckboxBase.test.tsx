import React from "react";
import { queryByAttribute, render } from "@origin-digital/ods-testing-library";
import { CheckboxBase } from "./CheckboxBase";

test("it renders an unchecked CheckboxBase", () => {
  const { container } = render(<CheckboxBase id="checkboxInput" />);

  expect(container).toMatchSnapshot();
});

test("it renders a checked CheckboxBase", () => {
  const { container } = render(<CheckboxBase id="checkboxInput" checked />);

  expect(container).toMatchSnapshot();
});

test("it renders a CheckboxBase in error", () => {
  const { container } = render(<CheckboxBase id="checkboxInput" error />);

  expect(container).toMatchSnapshot();
});

test("it renders a disabled CheckboxBase", () => {
  const { container } = render(<CheckboxBase id="checkboxInput" disabled />);

  expect(container).toMatchSnapshot();
});

describe("uncontrolled CheckboxBase", () => {
  test("it passes down id and data-id", () => {
    const { container } = render(
      <CheckboxBase data-id="checkboxRoot" id="mychk" />
    );

    const chkRoot = queryByAttribute("data-id", container, "checkboxRoot");
    const input = queryByAttribute("id", container, "mychk");

    expect(chkRoot).toBeDefined();
    expect(input?.tagName.toLowerCase()).toEqual("input");
  });

  test("it is clicked, it will set Mui-Checked prop", () => {
    const { container } = render(
      <CheckboxBase data-id="checkboxRoot" id="mychk" />
    );

    const chkRoot = queryByAttribute("data-id", container, "checkboxRoot");
    const input = queryByAttribute("id", container, "mychk");

    expect(chkRoot).not.toHaveClass("Mui-checked");

    input?.click();

    expect(chkRoot).toHaveClass("Mui-checked");
  });
});
