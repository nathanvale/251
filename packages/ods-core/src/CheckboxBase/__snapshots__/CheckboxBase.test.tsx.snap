// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a CheckboxBase in error 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-95 MuiCheckbox-root AbstractSwitchBase-error-root-87 AbstractSwitchBase-error-root-88 MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    data-id="checkboxBase"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-98"
        data-indeterminate="false"
        id="checkboxInput"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m2-2H3v18h18V3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;

exports[`it renders a checked CheckboxBase 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-52 MuiCheckbox-root AbstractSwitchBase-root-44 AbstractSwitchBase-root-45 MuiCheckbox-colorSecondary PrivateSwitchBase-checked-53 Mui-checked MuiIconButton-colorSecondary"
    data-id="checkboxBase"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        checked=""
        class="PrivateSwitchBase-input-55"
        data-indeterminate="false"
        id="checkboxInput"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 3H3v18h18V3zM10 17l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;

exports[`it renders a disabled CheckboxBase 1`] = `
<div>
  <span
    aria-disabled="true"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-138 MuiCheckbox-root AbstractSwitchBase-root-130 AbstractSwitchBase-root-131 MuiCheckbox-colorSecondary PrivateSwitchBase-disabled-140 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
    data-id="checkboxBase"
    tabindex="-1"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-141"
        data-indeterminate="false"
        disabled=""
        id="checkboxInput"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m2-2H3v18h18V3z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`it renders an unchecked CheckboxBase 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root AbstractSwitchBase-root-1 AbstractSwitchBase-root-2 MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    data-id="checkboxBase"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-12"
        data-indeterminate="false"
        id="checkboxInput"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m2-2H3v18h18V3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;
